File: c/raindrops/raindrops.c
#include "raindrops.h"

#include <string.h>
#include <stdio.h>

void convert(char result[], int drops) {
    if (drops % 3 == 0 || drops % 5 == 0 || drops % 7 == 0) {
        if (drops % 3 == 0) {
            strcat(result, "Pling");
        }
        if (drops % 5 == 0) {
            strcat(result, "Plang");
        }
        if (drops % 7 == 0) {
            strcat(result, "Plong");
        }
    } else {
        sprintf(result, "%d", drops);
    }
}

File: go/raindrops/raindrops.go
package raindrops

import "fmt"
import "strings"

func Convert(number int) string {
	if number%3 != 0 && number%5 != 0 && number%7 != 0 {
		return fmt.Sprintf("%d", number)
	}

	var b strings.Builder
	if number%3 == 0 {
		b.WriteString("Pling")
	}

	if number%5 == 0 {
		b.WriteString("Plang")
	}

	if number%7 == 0 {
		b.WriteString("Plong")
	}

	return b.String()
}
